cpp/src/arrow/compute/exec/aggregate_node.cc

```diff
@@ -60,17 +101,47 @@ void AggregatesToString(std::stringstream* ss, const Schema& input_schema,
   *ss << ']';
 }
 
+ExecBatch SelectBatchValues(const ExecBatch& batch, const std::vector<int>& ids) {
+  std::vector<Datum> values(ids.size());
+  for (size_t i = 0; i < ids.size(); i++) {
+    values[i] = batch.values[ids[i]];
+  }
+  return ExecBatch(std::move(values), batch.length);
+}
```

Comment:
Could we add a `Select` method to `ExecBatch` instead?

Reply:
Done.